chore: Doc update about incrementals supporting append destinations (#18033)
cloudquery/cloudquery: website/pages/docs/advanced-topics/managing-incremental-tables.md

When a sync runs on an incremental table, the table will first fetch the last known cursor state from the state backend, then resume syncing from that point. Incremental tables guarantee at-least-once delivery, which means that there should never be gaps in the data as a result of the cursor state being used, but there may be duplicates. If the destination uses `overwrite` or `overwrite-delete-stale` write mode, these duplicates will be handled automatically. But if the destination uses `append` mode, care will need to be taken to either exclude the duplicates at query time or to run a deduplication process on the destination. 

> In order to resume from a previous position, incremental tables store some state, known as the **cursor**. When using the CloudQuery CLI, the cursor state is stored in a **backend**. Any destination plugin can be used as a backend. This state destination is often the same destination data is being written to for the sync, but it doesn't need to be. In some destinations which support `append` mode, each state update will generate a new row (with a new version) in the state table and may need to be cleaned up periodically (this feature is not currently enabled for all `append`-only destinations). Destinations supporting `overwrite` mode don't have this issue.

Reviewer: ```suggestion
In order to resume from a previous position, incremental tables store some state, known as the **cursor**. When using the CloudQuery CLI, the cursor state is stored in a **backend**. Any destination plugin can be used as a backend. This state destination is often the same destination data is being written to for the sync, but it doesn't need to be. In some destinations that support only `append` mode, each state update will generate a new row (with a new version) in the state table and may need to be cleaned up periodically. Destinations supporting `overwrite` mode don't have this issue.
```
